python/mxnet/gluon/model_zoo/model_store.py

```diff
@@ -103,16 +106,16 @@ def get_model_file(name, root=os.path.join(base.data_dir(), 'models')):
 
     util.makedirs(root)
 
-    zip_file_path = os.path.join(root, file_name+'.zip')
+    temp_zip_file_path = os.path.join(root, file_name+random_uuid+'.zip')
```

Comment:
Do you want to use tempfile so it takes care of clean up automatically: https://docs.python.org/3/library/tempfile.html

Reply:
I believe tempfile is typically on a different filesystem, so there is no atomic rename operation to the target directory and filename available.

Reply:
@leezu What do you mean by a different filesystem? Regarding name, I think you can also specify a file name using the NamedTemporaryfile class?

Reply:
You're right, I overlooked the `dir` argument.